### ISSUE TITLE
Improve `unistd::Whence`

### ISSUE THREAD

**Issue by Susurrus** [nix-rust/nix#676]
Needs a few things here:
 * [ ] Whence needs to be documented
 * [ ] Hardcoded constants for `SEEK_DATA` and `SEEK_HOLE` need to be converted to use `libc` types.
 * [ ] `SeekData` and `SeekHole` should only be available on supported platforms.
 * [ ] Instead of having different enum values and libc values, just use the libc values directly for each enum variant. Maybe improve the `libc_enum!` to support variable renames to make this easy.

**Comment by ncaracci** [comment]
Is it okay to use the descriptions found on linux's lseek manpage?

**Comment by asomers** [comment]
For Linux-specific constants, yes.  But for standardized constants, it's usually better to use the OpenGroup's descriptions.  You can find them at http://pubs.opengroup.org/onlinepubs/9699919799/functions/lseek.html .

**Comment by ncaracci** [comment]
Would this be okay?
```
/// Directive that tells lseek how to reposition the offset of the file.
pub enum Whence {
    /// The file offset shall be set to offset bytes.
    SeekSet,
    /// The file offset shall be set to its current location plus offset.
    SeekCur,
    /// The file offset shall be set to the size of the file plus offset.
    SeekEnd,
    /// Adjust the file offset to the next location in the file greater than
    /// or equal to offset containing data.  If offset points to data, then 
    /// the file offset is set to offset.
    SeekData,
    /// Adjust the file offset to the next hole in the file greater than or
    /// equal to offset.  If offset points into the middle of a hole, then 
    /// the  file  offset is  set to offset.  If there is no hole past offset,
    /// then the file offset is adjusted to the end of the file (i.e., there
    /// is an implicit hole at the end of any file).
    SeekHole
}
```

**Comment by Susurrus** [comment]
@ncaracci Can you put this in a PR? It's easier to comment on these when I can put comments on specific lines. And you're definitely on the right track, so you shouldn't worry about this getting merged.

**Comment by ncaracci** [comment]
Ok, here's the PR #730.

**Comment by Susurrus** [comment]
Closed by 3d7d7ea5445ea0e2626822bd397dd3dad45ac8e6.

Thanks for your contribution @ncaracci!
